vortex-ipc/src/reader.rs

```diff
@@ -168,51 +167,67 @@ impl<'a, R: Read> StreamArrayReader<'a, R> {
             vortex_bail!("Indices must be positive")
         }
 
-        if self.row_offset != 0 {
```

Comment:
no longer necessary because take consumes the reader

Reply:
Caller can still call `reader.next()?; reader.take()` and the indices won't make sense.

Reply:
ahh good point